src/v/cluster/partition_manager.h

```diff
@@ -197,6 +199,7 @@ class partition_manager {
       _partition_recovery_mgr;
     ss::sharded<cloud_storage::remote>& _cloud_storage_api;
     ss::sharded<cloud_storage::cache>& _cloud_storage_cache;
+    ss::lw_shared_ptr<archival::configuration> _archival_conf;
```

Comment:
I've made this a pointer to const.  It's true that it could use bare references, but for objects that don't have interesting start/stop behaviors I find the ergonomics of a shared ptr nice, things like constructing objects in unit tests and passing the ptr into them instead of having to keep an object alive externally for them to refer to.